### PR TITLE
Return sync propose variant (default) together with new async version

### PR DESCRIPTION
**Pull request by zsluedem**
## Overview

Make propose in synchrony way like before.

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by zsluedem** [comment]
bors try

**Comment by zsluedem** [comment]
bors try

**Comment by nzpr** [comment]
@zsluedem could you change title and description? This enables sync version by default and adds async version as an option, right?

File: casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala [hunk 1]
```diff
@@ -51,7 +51,7 @@ object BlockAPI {
 
   def deploy[F[_]: Concurrent: EngineCell: Log: Span](
       d: Signed[DeployData],
-      triggerPropose: Option[Casper[F] => F[Option[Int]]]
+      triggerPropose: Option[Casper[F] => F[Option[BlockHash]]]
```
Comment (nzpr) [inline casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala]: Is this commit relevant?

Comment (zsluedem) [reply casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala]: Yes. What is the problem?

Comment (zsluedem) [reply casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala]: It is outdated. I am curious how this comes.

Comment (nzpr) [reply casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala]: shouldn't  https://github.com/rchain/rchain/pull/3352/commits/6dc22c81a8cb31f996583bd550158bf8dd3417d7  be squashed into https://github.com/rchain/rchain/pull/3352/commits/5fca71c44b71395a7c240ac9deb84dd0723b7d7a ?they are changing the same parts of the code, and first commit is basically overwritten? Or I miss smth?

Comment (zsluedem) [reply casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala]: I don't understand why you look at the outdated commit. I don't understand how you find the https://github.com/rchain/rchain/commit/6dc22c81a8cb31f996583bd550158bf8dd3417d7 here. Did you review before I assign you?

I suggest you just review whatever it is right now.

Comment (zsluedem) [reply casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala]: > Is this commit relevant?

Oh. Sorry.   That commit is now relevant. I don't know how you find that.

Comment (nzpr) [reply casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala]: I'm not sure - are you saying its relevant or not? Both commits are in this PR, and second one introduces async and sync versions. Why to have first commit that reverts current async to sync? Why not introduces async and sync versions right away but first revert to sync? It touches the same parts of the code, right?

Comment (zsluedem) [reply casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala]: Ok. I am going to squash it. We are playing words game which is not important

Comment (nzpr) [reply casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala]: I'm not sure its words game - I simply do not understand whats going on :) If one commit is changing a line, then another commit is changing the same line and nothing else - I believe its reasonable to not have these two separate. There should be only the last commit.

**Review by nzpr — APPROVED** [review]
Thanks, I believe now its easier to review

File: casper/src/main/scala/coop/rchain/casper/package.scala [hunk 2]
```diff
@@ -10,7 +10,7 @@ package object casper {
   type BlockProcessing[A]   = Either[BlockError, A]
   type ValidBlockProcessing = BlockProcessing[ValidBlock]
 
-  type ProposeFunction[F[_]] = (Casper[F] => F[Option[Int]])
+  type ProposeFunction[F[_]] = ((Casper[F], Boolean) => F[Option[Either[Int, BlockHash]]])
```
Comment (tgrospic) [inline casper/src/main/scala/coop/rchain/casper/package.scala:13]: Using nested `Option[Either[_,_]]` and other combinations is very hard to understand. I'm ok with this for now but we will need better solution.

**Review by tgrospic — APPROVED** [review]
LGTM

File: casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala (hunk 1 above)
Comment (zsluedem) [reply casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala]: > I'm not sure its words game - I simply do not understand whats going on :) If one commit is changing a line, then another commit is changing the same line and nothing else - I believe its reasonable to not have these two separate. There should be only the last commit.

It doesn't make sense to me, though.

**Comment by zsluedem** [comment]
bors merge

**Pull request merged**